src/info/manifest.c

```diff
@@ -383,6 +386,750 @@ manifestNewInternal(void)
+                buildDataSub.dbPath = regExpMatch(buildData.dbPathExp, manifestName);
+
+            storageInfoListP(
+                buildDataSub.storagePg, buildDataSub.pgPath, manifestBuildCallback, &buildDataSub, .sortOrder = sortOrderAsc);
+
+            break;
+        }
+
+        // Add files
+        // -------------------------------------------------------------------------------------------------------------------------
+        case storageTypeFile:
+        {
+            // Skip pg_internal.init since it is recreated on startup
+            if (strEqZ(info->name, PG_FILE_PGINTERNALINIT))
+            {
+                FUNCTION_TEST_RETURN_VOID();
+                return;
+            }
+
+            // Skip files in the root data path
+            if (strEq(buildData.manifestParentName, MANIFEST_TARGET_PGDATA_STR))
+            {
+                // Skip recovery files
+                if (((strEqZ(info->name, PG_FILE_RECOVERYSIGNAL) || strEqZ(info->name, PG_FILE_RECOVERYSIGNAL)) &&
+                        pgVersion >= PG_VERSION_12) ||
+                    ((strEqZ(info->name, PG_FILE_RECOVERYCONF) || strEqZ(info->name, PG_FILE_RECOVERYDONE)) &&
+                            pgVersion < PG_VERSION_12) ||
+                    // Skip temp file for safely writing postgresql.auto.conf
+                    (strEqZ(info->name, PG_FILE_POSTGRESQLAUTOCONFTMP) && pgVersion >= PG_VERSION_94) ||
+                    // Skip backup_label in versions where non-exclusive backup is supported
+                    (strEqZ(info->name, PG_FILE_BACKUPLABEL) && pgVersion >= PG_VERSION_96) ||
+                    // Skip old backup labels
+                    strEqZ(info->name, PG_FILE_BACKUPLABELOLD) ||
+                    // Skip running postmaster options
+                    strEqZ(info->name, PG_FILE_POSTMASTEROPTS) ||
+                    // Skip process id file to avoid confusing postgres after restore
+                    strEqZ(info->name, PG_FILE_POSTMASTERPID))
+                {
+                    FUNCTION_TEST_RETURN_VOID();
+                    return;
+                }
+            }
+
+            // Skip the contents of the wal path when online
+            if (buildData.online && strEq(buildData.manifestParentName, buildData.manifestWalName))
+            {
+                FUNCTION_TEST_RETURN_VOID();
+                return;
+            }
+
+            // Skip temp relations in db paths
+            if (buildData.dbPath && regExpMatch(buildData.tempRelationExp, info->name))
+            {
+                FUNCTION_TEST_RETURN_VOID();
+                return;
+            }
+
+            // Add file to manifest
+            ManifestFile file =
+            {
+                .name = manifestName,
+                .mode = info->mode,
+                .user = info->user,
+                .group = info->group,
+                .size = info->size,
+                .sizeRepo = info->size,
```

Comment:
Why are we defaulting repoSize to fileSize - it is not the same thing. Will this be updated by some other code somewhere?

Reply:
It is updated by manifestFileUpdate() when the file is copied/resumed/referenced.